syncer/syncer.go

```diff
@@ -716,6 +729,58 @@ func (s *Syncer) addCount(isFinished bool, queueBucket string, tp opType, n int6
 	}
 }
 
+func (s *Syncer) jobsIsEmpty() bool {
+	for idx := range s.jobs {
+		if len(s.jobs[idx]) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// updateReplicationLag calc syncer replication lag by job, it called after every batch dml job / one skip job / one ddl
+// job is flushed to target db.
+func (s *Syncer) updateReplicationLag(job *job, queueBucketName string) {
+	if job != nil {
+		var lag int64
+		switch job.tp {
+		case ddl:
+			failpoint.Inject("BlockSyncerUpdateDDLLag", func(v failpoint.Value) {
```

Comment:
I think one failpoint is enough. We can a value to job.tp and check whether `job.tp.String()==v`